extensions/src/Smithy.Identity.Abstractions/IIdentity.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+
+namespace Smithy.Identity.Abstractions
+{
+    /// <summary>
+    /// An identity is an entity within the SDK representing who the customer is.
+    /// For example, the customer's identity could be anonymouus, a token, a public / private key pair, etc...
+    /// </summary>
+    public abstract class IIdentity
```

Comment:
Since we are not actually an interface we should not start with the extra letter `I`. Since we are using abstract classes I would suggest for all of these abstract class uses the prefix `Base` or `Abstract`. My preference is `Base`.

Reply:
Renamed the class to `BaseIdentity`: https://github.com/aws/aws-sdk-net/pull/3330/commits/285e421c304ad6d5c21deb54e6f85ec374226275#diff-37d2151f0ebc200ab67497b246f32ca3d73c3a9d1a1ac999098f5229eb7ebdcf